utils/handlers/handlers.go

```diff
@@ -90,13 +99,20 @@ type AppHandler func(http.ResponseWriter, *http.Request) *AppError
 // ServeHTTP responds via the passed handler and handles returned errors
 func (fn AppHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	w.Header().Set("content-type", "application/json")
+	reqID := r.Header.Get(requestIDHeaderKey)
+	w.Header().Set(requestIDHeaderKey, reqID)
+	ctx := context.WithValue(r.Context(), RequestID, reqID)
+	r = r.WithContext(ctx)
```

Comment:
I would suggest following a `middleware` model here, much like `middleware.BearerToken`  I feel that this could be extended to handle other headers that we wish to propagate to the request context in the future.  It almost feels like a one-off when implemented in AppHandler.ServeHTTP.